src/core/image.js

```diff
@@ -0,0 +1,539 @@
+(function() {
+
+  function Queue() {
+
+    // callback storage
+    this._methods = [];
+
+    // reference to the response
+    this._response = null;
+
+    // all queues start off unflushed
+    this._flushed = false;
+
+  };
+
+  Queue.prototype = {
+
+    // adds callbacks to the queue
+    add: function(fn) {
+
+      // if the queue had been flushed, return immediately
+      if (this._flushed) {
+
+        // otherwise push it on the queue
+        fn(this._response);
+
+      } else {
+        this._methods.push(fn);
+      }
+
+    },
+
+    flush: function(resp) {
+
+      // flush only ever happens once
+      if (this._flushed) {
+        return;
+      }
+
+      // store the response for subsequent calls after flush()
+      this._response = resp;
+
+      // mark that it's been flushed
+      this._flushed = true;
+
+      // shift 'em out and call 'em back
+      while (this._methods[0]) {
+        this._methods.shift()(resp);
+      }
+
+    }
+
+  };
+
+  ImageModel = cdb.core.Model.extend({
+    defaults: {
+      format: "png",
+      zoom: 10,
+      center: [0, 0],
+      size:  [320, 240],
+      tiler_port: 80,
+      tiler_domain: "cartodb.com"
+    }
+  });
+
+  var Image = function() {
+
+    var self = this;
+
+    this.model = new ImageModel();
+
+    this.supported_formats = ["png", "jpg"];
+
+    this.defaults = {
+      tiler_domain: "cartodb.com",
+      tiler_port: "80"
+    };
+
+    this.available_basemaps = ["light_all", "light_nolabels", "dark_all", "dark_nolabels"];
+    this.options = _.defaults({
```

Comment:
this constructor is almost a copy for Map constructor, shouldn't we call Map constructor instead?